pyomo/common/collections/component_set.py

```diff
@@ -104,11 +104,11 @@ def discard(self, val):
     # plain dictionary mapping key->(type(val), id(val)) and
     # compare that instead.
     def __eq__(self, other):
-        if not isinstance(other, collections_Set):
+        if self is other:
+            return True
+        if not isinstance(other, collections_Set) or len(self) != len(other):
             return False
-        return set((type(val), id(val)) for val in self) == set(
-            (type(val), id(val)) for val in other
-        )
+        return all(id(key) in self._data for key in other)
```

Comment:
That is covered by the test above (`if len(self) != len(other)`): if their lengths are the same, then you only have to test one side (because neither can have duplicates).  I will break that test apart to make it a little more clear

Reply:
Ahh! Got it!